src/LaunchDarkly.EventSource/Internal/EventParser.cs

```diff
@@ -0,0 +1,307 @@
+﻿using System;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using LaunchDarkly.EventSource.Events;
+using LaunchDarkly.Logging;
+
+using static LaunchDarkly.EventSource.Internal.AsyncHelpers;
+
+namespace LaunchDarkly.EventSource.Internal
+{
+    /// <summary>
+    /// An internal class containing helper methods to parse Server Sent Event data.
+    /// </summary>
+    internal sealed class EventParser
```

Comment:
Again this is a pretty direct port of [the Java code](https://github.com/launchdarkly/okhttp-eventsource/blob/main/src/main/java/com/launchdarkly/eventsource/EventParser.java), but it is smaller because it doesn't yet include any of the "provide a stream for the caller to read event data incrementally" logic.